### PR TITLE
#108 adapt existing state of new contact to add modal

### PR DESCRIPTION
**Pull request by asmaNode**
**DEMO :**
https://user-images.githubusercontent.com/43316837/111963106-6fa98a00-8af3-11eb-829c-1e98d2c573fe.mp4

### CODE REVIEW

File: src/linagora.esn.contact/app/app.controllers.js [hunk 1]
```diff
@@ -35,6 +35,14 @@ angular.module('linagora.esn.contact')
...
     $scope.addressbookPath = '/addressbooks/' + $scope.bookId + '/' + $scope.bookName + '.json';
 
+    function _hideModal() {
+      $scope.$hide();
+    }
+    $scope.$on('$stateChangeStart', function(event, toState) {
+      if (toState.name !== 'contact.addressbooks.new') {
+        _hideModal();
+      }
+    });
```
Comment (rezk2ll) [inline src/linagora.esn.contact/app/app.controllers.js:45]: this listens to every state change in the application.

you can simply close the modal when leaving the state by adding 
```js
          onExit: function(modalInstance) {
            modalInstance.hide();
          }
```
to the state

File: src/linagora.esn.contact/app/app.routes.js [hunk 2]
```diff
@@ -69,14 +69,14 @@ require('./services/contact-configuration.service.js');
...
-              resolve: {
+          resolve: {
+            modalInstance: function($modal) {
+              return $modal({
+                template: require('./contact/create/contact-create.pug'),
+                controller: 'newContactController',
                 domain: routeResolver.session('domain'),
                 user: routeResolver.session('user')
-              }
+              });
```
Comment (rezk2ll) [inline src/linagora.esn.contact/app/app.routes.js:79]: you can use the following code to hide the modal when leaving this state
```
          onExit: function(modalInstance) {
            modalInstance.hide();
          }
```

File: src/linagora.esn.contact/app/contact/create/contact-create.pug [hunk 3]
```diff
@@ -1,7 +1,8 @@
-sub-header
-  contact-create-subheader
+.modal.full-screen(tabindex="-1", role="dialog")
```
Comment (rezk2ll) [inline src/linagora.esn.contact/app/contact/create/contact-create.pug:1]: the template is not part of the issue

Comment (asmaNode) [reply src/linagora.esn.contact/app/contact/create/contact-create.pug:1]: yes , i know i just added it to test the behaviour of modal

File: src/linagora.esn.contact/app/app.routes.js (hunk 2 above)
Comment (asmaNode) [reply src/linagora.esn.contact/app/app.routes.js:82]: the some comment ?

**Review by rezk2ll — DISMISSED** [review]

**Review by rezk2ll — APPROVED** [review]

**Pull request merged**